### PR TITLE
[ONNX] Bool inputs to index_put updated symbolic

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Cases with bool inputs to index_put nodes were handled for tracing purposes. This PR adds support for similar situations in scripting

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ static void ReplaceIndexPutWithMaskedScatter(Block* b) {
 void PreprocessForONNX(std::shared_ptr<Graph>& graph) {
   FuseWithListUnpack(graph->block());
   ReplaceAddWithConcat(graph->block());
-  ReplaceIndexPutWithMaskedScatter(graph->block());
+  // ReplaceIndexPutWithMaskedScatter(graph->block());
```
Comment (KsenijaS) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:244]: nit remove comment

File: torch/onnx/symbolic_opset11.py [hunk 2]
```diff
@@ -80,6 +80,13 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
         ]
         index = g.op("Concat", *indices_list, axis_i=-1)
     else:
+        bool_inp = list(index.node().inputs())[0]
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:83]: Could you add a few comments maybe moving the ir snippet from the pass down to here? I'm confused why need to check input of index node.

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -80,6 +80,13 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
...
+        if bool_inp.type().scalarType() == 'Bool':
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:84]: please check first if `type()` is not None.

File: torch/onnx/symbolic_opset11.py [hunk 4]
```diff
@@ -80,6 +80,13 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
...
+            if values.node().kind() == 'onnx::Constant':
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:85]: why the requirement on constant?

From below looks like trying to get rank info. You can achieve that through `values.type().dim()`, remember to check `values.type()` is not None.

File: torch/onnx/symbolic_opset11.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py:83]: Moved ir snippet to opset files, I'm unsure how to structure the example

File: torch/onnx/symbolic_opset11.py [hunk 5]
```diff
@@ -80,6 +80,32 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
...
+        # after the change
+        # graph(%0 : Float(2:4, 2:2, 2:1, requires_grad=0, device=cpu)):
+        # %46 : Float(requires_grad=0, device=cpu) = prim::Constant[value={5}]()
+        # %mask.1 : Float(2:4, 2:2, 2:1, requires_grad=0, device=cpu) =
+        # %23 : Float(requires_grad=0, device=cpu) = prim::Constant[value={1}]()
+        # %24 : bool = prim::Constant[value=0]()
+        # %49 : Tensor = aten::masked_fill(%mask.1, %21, %23)
+        bool_inp = list(index.node().inputs())[0]
+        if bool_inp.type() is not None and bool_inp.type().scalarType() is not None:
+            if bool_inp.type().scalarType() == 'Bool':
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:103]: I think this line can replace the above `bool_inp.type().scalarType() is not None`

File: torch/onnx/symbolic_opset11.py (hunk 2 above)
Comment (BowenBao) [reply torch/onnx/symbolic_opset11.py:101]: do we need to copy masked_scatter comments as well?

File: torch/onnx/symbolic_opset11.py [hunk 6]
```diff
@@ -80,6 +80,32 @@ def index_put(g, self, indices_list_value, values, accumulate=False):
...
+        # %24 : bool = prim::Constant[value=0]()
+        # %mask : Float(2:4, 2:2, 2:1) = aten::index_put_(%mask.1, %22, %23, %24)
+        # 
+        # after the change
+        # graph(%0 : Float(2:4, 2:2, 2:1, requires_grad=0, device=cpu)):
+        # %46 : Float(requires_grad=0, device=cpu) = prim::Constant[value={5}]()
+        # %mask.1 : Float(2:4, 2:2, 2:1, requires_grad=0, device=cpu) =
+        # %23 : Float(requires_grad=0, device=cpu) = prim::Constant[value={1}]()
+        # %24 : bool = prim::Constant[value=0]()
+        # %49 : Tensor = aten::masked_fill(%mask.1, %21, %23)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:100]: Please update the IR graph above since it is no longer the same now that this is at onnx conversion.

**Review by BowenBao — APPROVED** [review]
LG, thanks.

**Review by bzinodev — APPROVED** [review]
